// instrument.proto

syntax = "proto3";

service InstrumentService {
  rpc GetStatus (StatusRequest) returns (StatusResponse);
  rpc SetChannel (ChannelRequest) returns (ChannelResponse);
  rpc ListDevices (Empty) returns (DeviceListResponse);
  rpc ConnectDevice (DeviceRequest) returns (ConnectionResponse);
  rpc DisconnectDevice (Empty) returns (ConnectionResponse);
  rpc InitializeVISA (Empty) returns (ConnectionResponse);
}

message Empty {}

message StatusRequest {}
message StatusResponse {
  string status = 1;
}

message ChannelRequest {
  int32 channel = 1;
  float voltage = 2;
  float current = 3;
}

message ChannelResponse {
  bool success = 1;
  string message = 2;
}

message DeviceListResponse {
  repeated string devices = 1;
}

message DeviceRequest {
  string address = 1;
}

message ConnectionResponse {
  bool success = 1;
  string message = 2;
}

